std/algorithm/comparison.d

```diff
@@ -637,6 +647,12 @@ do
     assert(x.clamp(lo, hi).y == 42);
 }
 
+// https://issues.dlang.org/show_bug.cgi?id=23268
+@safe pure nothrow @nogc unittest
+{
+    static assert(__traits(compiles, clamp(short.init, short.init, cast(const)short.init)));
```

Comment:
Just put the expression in the unittest instead of `assert(__traits(compiles, ...))` so the error isn't gagged when the test fails.

Reply:
The phobos way is to put compilable tests into `traits(compiles)` blocks. That way the intent is clear.

Reply:
Also, we don't care about this code being run and generating a clamp instance that makes it to the object file.

Reply:
You can clarify the intent with a comment

Reply:
There is a number of ways to do this, buy I prefer the to be consistent with the rest of the test suite.